### ISSUE TITLE
build problem with v0.12.1

### ISSUE THREAD

**Issue by silvio** [vector-im/riot-web#4863]
### Description

With latest release I got it not managed to build it via docker. It's not a docker problem.

```
> riot-web@0.12.1 build:bundle /riot-web                                                                                                                                                                                                                                                                                                                                     
> cross-env NODE_ENV=production webpack -p --progress --bail                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                                                                             
 54% 655/874 build modulesModuleNotFoundError: Module not found: Error: Cannot resolve 'file' or 'directory' ../../wrappers/withMatrixClient in /riot-web/node_modules/matrix-react-sdk/lib/components/structures                                                                                                                                                            
    at /riot-web/node_modules/webpack/lib/Compilation.js:229:38                            
    at onDoneResolving (/riot-web/node_modules/webpack/lib/NormalModuleFactory.js:29:20)   
    at /riot-web/node_modules/webpack/lib/NormalModuleFactory.js:85:20                     
    at /riot-web/node_modules/async/lib/async.js:726:13                                    
    at /riot-web/node_modules/async/lib/async.js:52:16                                     
    at done (/riot-web/node_modules/async/lib/async.js:241:17)                             
    at /riot-web/node_modules/async/lib/async.js:44:16                                     
    at /riot-web/node_modules/async/lib/async.js:723:17                                    
    at /riot-web/node_modules/async/lib/async.js:167:37                                    
    at /riot-web/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:24:19              
    at onResolved (/riot-web/node_modules/enhanced-resolve/lib/Resolver.js:38:18)          
    at /riot-web/node_modules/enhanced-resolve/lib/Resolver.js:127:10                      
    at /riot-web/node_modules/enhanced-resolve/lib/Resolver.js:191:15                      
    at applyPluginsParallelBailResult.createInnerCallback.log (/riot-web/node_modules/enhanced-resolve/lib/Resolver.js:110:4)                                                         
    at loggingCallbackWrapper (/riot-web/node_modules/enhanced-resolve/lib/createInnerCallback.js:21:19)                                                                              
    at /riot-web/node_modules/tapable/lib/Tapable.js:134:6                                 
    at Tapable.<anonymous> (/riot-web/node_modules/enhanced-resolve/lib/DirectoryDescriptionFilePlugin.js:24:12)                                                                      
    at Storage.finished (/riot-web/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)                                                                                  
    at ReadFileContext.callback (/riot-web/node_modules/graceful-fs/graceful-fs.js:78:16)  
    at FSReqWrap.readFileAfterOpen [as oncomplete] (fs.js:365:13)                          
resolve file                                                                               
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient doesn't exist      
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient.webpack.js doesn't exist                                                                                      
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient.web.js doesn't exist                                                                                          
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient.js doesn't exist   
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient.json doesn't exist 
resolve directory                                                                          
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient doesn't exist (directory default file)                                                                        
  /riot-web/node_modules/matrix-react-sdk/lib/wrappers/withMatrixClient/package.json doesn't exist (directory description file)                                                       
 70% 1755/1755 build modules                                                               
npm ERR! Linux 4.12.3-1-ARCH                                                               
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "build:bundle"                          
npm ERR! node v6.9.5                                                                       
npm ERR! npm  v3.10.10                                                                     
npm ERR! code ELIFECYCLE                                                                   
npm ERR! riot-web@0.12.1 build:bundle: `cross-env NODE_ENV=production webpack -p --progress --bail`                                                                                   
npm ERR! Exit status 1                                                                     
npm ERR!                                                                                   
npm ERR! Failed at the riot-web@0.12.1 build:bundle script 'cross-env NODE_ENV=production webpack -p --progress --bail'.                                                              
npm ERR! Make sure you have the latest version of node.js and npm installed.               
npm ERR! If you do, this is most likely a problem with the riot-web package,               
npm ERR! not with npm itself.                                                              
npm ERR! Tell the author that this fails on your system:                                   
npm ERR!     cross-env NODE_ENV=production webpack -p --progress --bail                    
npm ERR! You can get information on how to open an issue for this project with:            
npm ERR!     npm bugs riot-web                                                             
npm ERR! Or if that isn't available, you can get their info via:                           
npm ERR!     npm owner ls riot-web                                                         
npm ERR! There is likely additional logging output above.                                  

npm ERR! Please include the following file with any support request:                       
npm ERR!     /riot-web/npm-debug.log                                                       
```

### Steps to reproduce

you need `docker` installed.

* git clone https://github.com/silvio/matrix-riot-docker.git
* cd matrix-riot-docker
* docker build -t riot-docker-buildtest .

### Version information

Version with this problem:

* commit: vector-im/riot-web@a65196ee8c91925873196d13d6299b700c79c339 
* tag: v0.12.1
* tag: 0.12.0-rc.2

Used tools:

* node v6.9.5                                                                       
* npm  v3.10.10

**Comment by silvio** [comment]
Version `0.12.0-rc.1` doesn't have this problem.

**Comment by richvdh** [comment]
This appears to be because the release of matrix-react-sdk is faulty and doesn't include withMatrixClient.js

**Comment by tchncs** [comment]
ah good i am not alone :cat2:

**Comment by DeX77** [comment]
No docker. Same here.

**Comment by Nostradamos** [comment]
Encountered this error yesterday and fixed it by doing an `npm install directory` in the soft linked matrix-react-sdk folder.

**Comment by DeX77** [comment]
Tried the workaround from @Nostradamos. Doesn't help.

**Comment by silvio** [comment]
@dbkr and @richvdh it looks like this problem is fixed with v0.12.2, at least for me.

**Comment by DeX77** [comment]
And here it is again :(
